src/v/cluster/partition.cc

```diff
@@ -202,8 +204,25 @@ ss::future<std::error_code> partition::prefix_truncate(
           _raft->ntp());
         co_return make_error_code(cluster::errc::feature_disabled);
     }
-    co_return co_await _log_eviction_stm->truncate(
-      truncation_offset, deadline, _as);
+    vlog(
+      clusterlog.info,
+      "Truncating {} to redpanda offset {} kafka offset {}",
+      _raft->ntp(),
+      truncation_offset,
+      kafka_truncation_offset);
+    if (truncation_offset != model::offset{}) {
+        auto err = co_await _log_eviction_stm->truncate(
+          truncation_offset, deadline, _as);
+        if (err) {
+            co_return err;
+        }
+    }
+    // TODO(awong): have archival listen on log_eviction_stm.
```

Comment:
I see this behavior modified in subsequent commits, maybe you can omit it altogether or mention in the commit message how this is temporary

Reply:
Yeah, I didn't scrub the history here too hard, given it progressed somewhat incrementally.